server/src/main/java/org/elasticsearch/index/mapper/XContentDataHelper.java

```diff
@@ -235,9 +218,17 @@ private static DocumentParserContext cloneDocumentParserContext(
         XContentParserConfiguration configuration,
         XContentBuilder builder
     ) throws IOException {
-        DocumentParserContext subcontext = context.switchParser(
-            XContentHelper.createParserNotCompressed(configuration, BytesReference.bytes(builder), context.parser().contentType())
+        XContentParser newParser = XContentHelper.createParserNotCompressed(
+            configuration,
+            BytesReference.bytes(builder),
+            context.parser().contentType()
         );
+        if (context.isDotExpandingEnabled()) {
+            // if we did dot expanding originally we need to continue to do so when we replace the parser
+            newParser = DotExpandingXContentParser.expandDots(newParser, context.path());
```

Comment:
Can we just use `context.path()` here? Or do we need to create a new one? The path no longer align with fields new parser, I think?

Reply:
It should be the same, this is the parser that we return for "regular" indexing so it should be at the same point as before the clone.